### PR TITLE
refactor: upgrade go version to 1.18.6

### PR DESCRIPTION
**Pull request by rajatgupta24**
This PR updates the go to 1.18.4.
This PR also updates the `lint` phony in Makefile.

related to: https://github.com/jenkins-x/jx/issues/8283

- [x] Change the version in `go.mod` file.
- [x] Run `go mod tidy` and ensure that it does not error out.
- [x] Change `GO_VERSION` in `makefile` to the version you are upgrading.
- [x] Check if there are any references to an older version and replace those.
- [x] Verify that
  - [x] The build works by running `make build`.
  - [x] Tests are working, by running `make test`.
  - [x] Linting checks are working by running `make lint`.

### CODE REVIEW

**Comment by rajatgupta24** [comment]
There's a warning while linting, might want to take a look at it.
```
WARN [linters context] structcheck is disabled because of go1.18. You can track the evolution of the go1.18 support by following the https://github.com/golangci/golangci-lint/issues/2649.
```

File: .github/workflows/jenkins-x-release.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
         GITHUB_TOKEN: ${{ secrets.GIT_BOT_TOKEN }}
         VERSION: ${{ steps.prep.outputs.version }}
       name: release-binary
-      uses: docker://golang:1.17.11
+      uses: docker://golang:1.18
```
Comment (ankitm123) [inline .github/workflows/jenkins-x-release.yaml:25]: change it to 1.18.6

File: .github/workflows/jenkins-x/upload-binaries.sh [hunk 2]
```diff
@@ -22,7 +22,7 @@ git push origin v$VERSION
 export BRANCH=$(git rev-parse --abbrev-ref HEAD)
 export BUILDDATE=$(date)
 export REV=$(git rev-parse HEAD)
-export GOVERSION="1.17.11"
+export GOVERSION="1.18"
```
Comment (ankitm123) [inline .github/workflows/jenkins-x/upload-binaries.sh:25]: 1.18.6

File: Makefile [hunk 3]
```diff
@@ -8,7 +8,7 @@ REV := $(shell git rev-parse --short HEAD 2> /dev/null || echo 'unknown')
 ROOT_PACKAGE := github.com/$(ORG_REPO)
 BRANCH     := $(shell git rev-parse --abbrev-ref HEAD 2> /dev/null  || echo 'unknown')
 BUILD_DATE := $(shell date +%Y%m%d-%H:%M:%S)
-GO_VERSION := 1.17.11
+GO_VERSION := 1.18
```
Comment (ankitm123) [inline Makefile:11]: Replace with section from jx-pipeline: https://github.com/jenkins-x-plugins/jx-pipeline/blob/89593c6e1e9d2c01b8a26296f53b6fdedcc48410/Makefile#L16-L22

File: hack/linter.sh [hunk 4]
```diff
@@ -9,38 +9,14 @@ fi
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
-if ! [ -x "$(command -v golangci-lint)" ]; then
-	echo "Installing GolangCI-Lint"
-	${DIR}/install_golint.sh -b $GOPATH/bin v1.20.0
+linterVersion="$(golangci-lint --version | awk '{print $4}')"
+
+expectedLinterVersion=1.46.2
```
Comment (ankitm123) [inline hack/linter.sh:14]: 1.47.3

**Review by ankitm123 — APPROVED** [review]

**Pull request merged**